client/fs_endpoint.go

```diff
@@ -177,11 +177,20 @@ func (f *FileSystem) stream(conn io.ReadWriteCloser) {
 		handleStreamResultError(allocIDNotPresentErr, pointer.Of(int64(400)), encoder)
 		return
 	}
-	alloc, err := f.c.GetAlloc(req.AllocID)
+
+	ar, err := f.c.getAllocRunner(req.AllocID)
 	if err != nil {
 		handleStreamResultError(structs.NewErrUnknownAllocation(req.AllocID), pointer.Of(int64(404)), encoder)
 		return
 	}
+	if ar.IsDestroyed() {
+		handleStreamResultError(
+			fmt.Errorf("allocation %s destroyed from client", req.AllocID),
+			pointer.Of(int64(404)),
```

Comment:
```suggestion
			pointer.Of(int64(http.StatusNotFound)),
```

Reply:
Fixed these and also others in the same file 👍